src/internal/js/mod.rs

```diff
@@ -307,10 +313,26 @@ impl JsString {
         }
     }
 
+    pub fn ucs2_value(self) -> Vec<u16> {
```

Comment:
> V8 can use also UTF-8 and Latin-1 as internal representations for the string, which would mean that sometimes this will introduce transforms to UCS-2

That's true but it can be avoided with the IsOneByte() and ContainsOnlyOneByte() methods.

> Can't we instead construct strings in V8 directly from UTF-8 representation that Rust provides?

Yes, if I understand you correctly.  V8 has the concept of external strings where the storage for the string lives outside the JS heap.  Node uses them to avoid copying its built-in scripts.

They have some overhead involved compared to regular strings though, so it's only worth it for larger strings.  We established empirically that the turning point is around the 1 MB mark.